lib/cog.ex

```diff
@@ -18,13 +18,18 @@ defmodule Cog do
     case Supervisor.start_link(children, opts) do
       {:ok, top_sup} ->
         # Verify the latest schema migration after starting the database worker
-        {sm_status, sm_message} = verify_schema_migration()
-        log_message(sm_status, sm_message)
-        if sm_status == :error do
-          abort_cog()
-        else
-          {:ok, top_sup}
-        end
+        #
+        # NOTE: This _may_ cause issues in the future, since we
+        # currently announce and install the embedded bundle during
+        # startup (i.e., before doing the schema migration check). If
+        # at some point we were to automatically upgrade the embedded
+        # bundle in such a way that required a database migration
+        # beforehand, the resulting error might be confusing.
+        #
+        # Consider doing this in a process that runs after the Repo
+        # comes up, but before anything else is done
+        :ok = verify_schema_migration!
```

Comment:
Here's an example:

```

=INFO REPORT==== 28-Mar-2016::10:29:54 ===
    application: logger
    exited: stopped
    type: temporary
** (Mix) Could not start application cog: exited in: Cog.start(:normal, [])
    ** (EXIT) an exception was raised:
        ** (RuntimeError) The database schema is at version 20160325211544, but the latest schema is 20160322222933. Please perform a migration and restart Cog.
            (cog) lib/cog.ex:117: Cog.verify_schema_migration!/0
            (cog) lib/cog.ex:31: Cog.start/2
            (kernel) application_master.erl:273: :application_master.start_it_old/4
make: *** [run] Error 1
```
Though I'm just noticing that the "have" version doesn't appear to correspond to any migration version when you rollback (which I did here... Hmmm...)